### ISSUE TITLE
iOS MSAL : Authority validation is not supported for this type of authority

### ISSUE THREAD

**Issue by alex-dokienko** [nodriver-ai/flutter_azure_b2c#21]
good day, 

when I'm using my authorityUrl in this format
https://{tenant}.b2clogin.com/{tenant}.onmicrosoft.com/{policy}

I'm getting the error on **policyTriggerInteractive** callback
"MSALErrorDescriptionKey" : "Authority validation is not supported for this type of authority"	
"MSALInternalErrorCodeKey" : Int64(-42008)
which stackoverflow say knowAuthorities should be properly set

But then I found that Microsoft [guidelines](https://learn.microsoft.com/en-us/azure/active-directory/develop/config-authority?WT.mc_id=Portal-Microsoft_AAD_RegisteredApps#b2c) says authority for iOS devices should be defined in this format
https://login.microsoftonline.com/tfp/{tenant}.onmicrosoft.com/{policy}

so when I changed to it (with some other tweaking to the plugin code to support new url format) I'm getting other error
[0]	(null)	"MSALErrorDescriptionKey" : "forbidden"	
[1]	(null)	"MSALHTTPResponseCodeKey" : "403"	
[3]	(null)	"MSALInternalErrorCodeKey" : Int64(-42007)	

I don't know which format is required and which direction should I go
I'd really appreciate any feedback.

**Comment by redoz** [comment]
@alex-dokienko Did you ever manage to figure this one out?

**Comment by alex-dokienko** [comment]
@redoz 
yes, I eventually resolved the issue, had to fork and managed to make it work with tfp url format
https://github.com/alex-dokienko/flutter_azure_b2c
